### PR TITLE
Improve herbstclient output passthrough in tests

### PR DESCRIPTION
**Pull request by dnnr**
This removes the python version selection from .travis.yml because
it doesn't work as expected, so no f-strings for us.

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -48,9 +48,13 @@ def _checked_call(self, cmd, expect_success=True):
                                   )
         except subprocess.TimeoutExpired:
             self.hlwm_process.investigate_timeout('calling ' + str(args))
-        print(list(args))
-        print(proc.stdout)
-        print(proc.stderr, file=sys.stderr)
+
+        outcome = 'succeeded' if proc.returncode == 0 else 'failed'
```
Comment (The-Compiler) [inline tests/conftest.py]: Would it be useful to know the exact return code?

File: tests/conftest.py [hunk 2]
```diff
@@ -48,9 +48,13 @@ def _checked_call(self, cmd, expect_success=True):
...
+        allout = proc.stdout + proc.stderr
+        if allout != '':
```
Comment (The-Compiler) [inline tests/conftest.py]: Can be simplified to `if allout:`

**Comment by dnnr** [comment]
This requires the fix provided by #219

**Comment by t-wissmann** [comment]
I don't get the issue here and the python error message does not help at all:
```
10.36s$ tox -- -n auto
py3 create: /home/travis/build/herbstluftwm/herbstluftwm/.tox/py3
py3 installdeps: pytest-xvfb, pytest-xdist
py3 installed: apipkg==1.5,atomicwrites==1.2.1,attrs==18.2.0,EasyProcess==0.2.3,execnet==1.5.0,more-itertools==4.3.0,pathlib2==2.3.3,pkg-resources==0.0.0,pluggy==0.8.0,py==1.7.0,pytest==4.0.1,pytest-forked==0.2,pytest-xdist==1.25.0,pytest-xvfb==1.1.0,PyVirtualDisplay==0.2.1,six==1.12.0
py3 runtests: PYTHONHASHSEED='1163509601'
py3 runtests: commands[0] | /home/travis/build/herbstluftwm/herbstluftwm/.tox/py3/bin/python -m pytest -n auto
ImportError while loading conftest '/home/travis/build/herbstluftwm/herbstluftwm/tests/conftest.py'.
  File "/home/travis/build/herbstluftwm/herbstluftwm/tests/conftest.py", line 55
    print(f'Client command {args} {outcome} with output:\n{allout}')
                                                                  ^
SyntaxError: invalid syntax
ERROR: InvocationError: '/home/travis/build/herbstluftwm/herbstluftwm/.tox/py3/bin/python -m pytest -n auto'
___________________________________ summary ____________________________________
ERROR:   py3: commands failed
The command "tox -- -n auto" exited with 1.
```

**Comment by t-wissmann** [comment]
(And tox runs without any issues here locally...)

**Comment by dnnr** [comment]
I found the actual mistake on my end. We now get to have Python 3.6 for test execution and everything is :100:.

**Pull request merged**